classes/local/cleaner/cleaner.php

```diff
@@ -158,34 +158,40 @@ private function clean_category() {
     }
 
     private function clean_course_by_id() {
-        global $DB;
-
         if (empty($this->params['id'])) {
-            return false;
+            return null;
         }
 
-        $slug = $DB->get_field('course', 'shortname', ['id' => $this->params['id']]);
-        $newpath = '/course/'.urlencode($slug);
+        $course = get_course($this->params['id']);
+
+        $newpath = '/course/' . urlencode($course->shortname);
         if ($this->check_path_allowed($newpath)) {
             $this->path = $newpath;
             unset($this->params['id']);
             clean_moodle_url::log("Rewrite course");
         }
-        return true;
+
+        return $course;
     }
 
     private function clean_course_by_name() {
+        global $DB;
+
         if (empty($this->params['name'])) {
             return false;
         }
 
-        $newpath = '/course/'.urlencode($this->params['name']);
+        $courseid = $DB->get_field('course', 'id', ['shortname' => $this->params['name']]);
+        $course = get_course($courseid);
```

Comment:
I need to get course data to check the format to handle the URL.

The second call should not be a problem, but I haven't found a good way (API) to translate a course shortname into id.